File: frontend/src/app/features/core/commission-setting/commission-setting-dto.ts
export interface CommissionSettingDto {
  id: number,
  partnerId: number | null,
  transactionId: number | null,
  sellerPercent: number,
  buyerPercent: number | null,
  director1Id: number | null,
  director1Percent: number | null,
  director2Id: number | null,
  director2Percent: number | null,
  director3Id: number | null,
  director3Percent: number | null,
  referralId: number | null,
  referralPercent: number | null
}

File: backend/src/main/java/hu/ibc/ibcpartners/core/dto/CommissionSettingDto.java
package hu.ibc.ibcpartners.core.dto;

import jakarta.validation.constraints.NotNull;

public record CommissionSettingDto(
        @NotNull Long id,
        Long partnerId,
        Long transactionId,
        @NotNull Integer sellerPercent,
        Integer buyerPercent,
        Long director1Id,
        Integer director1Percent,
        Long director2Id,
        Integer director2Percent,
        Long director3Id,
        Integer director3Percent,
        Long referralId,
        Integer referralPercent) {
}
